### PR TITLE
Use the right types for enums in Accessors method signatures.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
We're using the underlying integer types right now.  Use the right enum types.

The change to use a "using" declaration for the NumericAttributeTraits
is just to make the code more readable when the template arg is long.
There are no other changes there.

#### Problem
See above.

#### Change overview
See above.

#### Testing
Code compiles, once I fixed some bugs in the door lock code that this flushed out.

### CODE REVIEW

File: src/app/clusters/door-lock-server/door-lock-server.cpp [hunk 1]
```diff
@@ -94,12 +94,12 @@ bool DoorLockServer::SetActuatorState(chip::EndpointId endpointId, bool newActua
     return false;
```
Comment (vivien-apple) [inline src/app/clusters/door-lock-server/door-lock-server.cpp:94]: Not at all your PR. But I have to admit that I'm confused by some of those methods returning `true`, others returning `false` and one of them returning `status`...

**Review by vivien-apple — APPROVED** [review]

**Review by jmeg-sfy — APPROVED** [review]

File: src/app/zap-templates/templates/app/attributes/Accessors.zapt [hunk 2]
```diff
@@ -26,10 +27,10 @@ namespace Attributes {
 {{else if (canHaveSimpleAccessors this)}}
 namespace {{asUpperCamelCase label}} {
 EmberAfStatus Get(chip::EndpointId endpoint, {{accessorGetterType this}} value); // {{type}} {{isArray}}
-EmberAfStatus Set(chip::EndpointId endpoint, {{asUnderlyingZclType type}} value);
+EmberAfStatus Set(chip::EndpointId endpoint, {{zapTypeToEncodableClusterObjectType type ns=parent.name forceNotNullable=true forceNotOptional=true}} value);
 {{#if isNullable}}
 EmberAfStatus SetNull(chip::EndpointId endpoint);
-EmberAfStatus Set(chip::EndpointId endpoint, const DataModel::Nullable<{{asUnderlyingZclType type}}> & value);
+EmberAfStatus Set(chip::EndpointId endpoint, {{zapTypeToEncodableClusterObjectType type ns=parent.name isArgument=true forceNotOptional=true}} value);
```
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply src/app/zap-templates/templates/app/attributes/Accessors.zapt:33]: The big problem is that here we might have an enum type, or we might have a plain integer/float type.  So just throwing the namespace in the template is not really right.

Now what we could try doing is making the namespace handling smarter in some way so you can specify "just include the part after 'chip::app'" or "just include the part after 'Clusters'"....    I will think about it as a followup.

File: examples/all-clusters-app/esp32/main/main.cpp [hunk 3]
```diff
@@ -241,9 +241,10 @@ class EditAttributeListModel : public ListScreen::Model
             ESP_LOGI(TAG, "name and cluster: '%s' (%s)", name.c_str(), cluster.c_str());
             if (name == "State" && cluster == "Lock")
             {
+                using namespace chip::app::Clusters;
```
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply examples/all-clusters-app/esp32/main/main.cpp:244]: Yeah, I basically made the minimal change that would preserve my sanity while not messing too much with what the authors of this file seem to prefer.   I'll do a followup to move this up.

File: src/app/clusters/door-lock-server/door-lock-server.h [hunk 4]
```diff
@@ -39,8 +39,8 @@ class DoorLockServer
     void InitServer(chip::EndpointId endpointId);
 
     bool SetLockState(chip::EndpointId endpointId, chip::app::Clusters::DoorLock::DlLockState newLockState);
-    bool SetActuatorState(chip::EndpointId endpointId, bool newActuatorState);
-    bool SetDoorState(chip::EndpointId endpointId, chip::app::Clusters::DoorLock::DlLockState newDoorState);
+    bool SetActuatorEnabled(chip::EndpointId endpointId, bool newActuatorState);
+    bool SetDoorState(chip::EndpointId endpointId, chip::app::Clusters::DoorLock::DlDoorState newDoorState);
```
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply src/app/clusters/door-lock-server/door-lock-server.h:43]: Compiler caught it, once the types were fixed!

File: src/app/clusters/door-lock-server/door-lock-server.cpp (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/clusters/door-lock-server/door-lock-server.cpp:94]: I think it's still in the early stage of being developed....

**Review by andy31415 — APPROVED** [review]

**Pull request merged**